andela/ah-backend-dojo
# #165273593 Fix : authenticated users views shouldn't increase views on article

#### What does this PR Do?
Fixes the bug in the article read statistics.

#### Description of the task to be completed?

Refactor unit test for reading stats.
Refactor article read stats functionality

#### Any background context you want to provide?
A user can contribute to reading starts(views and reads) of a specific article once
An anonymous user does not contribute to the read stats (views and reads) of an article
I also found it necessary to add the endpoint for adding a user's read of an article.
A user cannot contribute to the views or reads of an article they published.

#### How should this be manually tested?
View an article you have created and check on the statistics. They should not change
View an article created by another user and check on its statistics again, the read stats should increase by one for the first view but remain constant for the other views.
Endpoint for increasing read stats of an article `POST [host]/api/articles/1/read_stat/`

#### What are the relevant pivotal tracker stories
[#165273593](https://www.pivotaltracker.com/story/show/165273593)
